src/utils.cpp

```diff
@@ -2247,7 +2247,7 @@ bool CacheableStatus::serialize(std::string* data)
     return const_cast<const CacheableStatus*>(this)->serialize(*data);
 }
 
-std::shared_ptr<CacheableStatus> CacheableStatus::unserialize(class MegaClient *client, const std::string& data)
+CacheableStatus* CacheableStatus::unserialize(class MegaClient *client, const std::string& data)
```

Comment:
we probably want to return nullptr in the fail cases

Reply:
Totally agree. I wonder why the compiler doesn't complain. A pointer to `CacheableStatus` shouldn't be infered from `{}`.
I'll fix it.